src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/batch/ConvertReceiveTypeWhenCallStepExecutionMethodTest.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.batch;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+import org.openrewrite.test.TypeValidation;
+
+import static org.openrewrite.java.Assertions.java;
+
+class ConvertReceiveTypeWhenCallStepExecutionMethodTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.parser(JavaParser.fromJavaVersion().classpathFromResources(new InMemoryExecutionContext(), "spring-batch-core-4", "spring-batch-infrastructure", "spring-beans", "mockito-core-5.+"))
+          .recipe(new ConvertReceiveTypeWhenCallStepExecutionMethod());
+    }
+
+    @DocumentExample
+    @Test
+    void test1() {
+        //language=java
+        rewriteRun(
+          spec -> spec.typeValidationOptions(TypeValidation.none()),
```

Comment:
Pointing this out mostly as a reminder to myself in a more thorough review that ideally we remove the need for these ignores before we merge, as it breaks the ability to chain recipes together for larger upgrades when the types go missing.
```suggestion
```